### ISSUE TITLE
Cancel PartitionBackfill Entirely

### ISSUE THREAD

**Issue by ruizh22** [dagster-io/dagster#14188]
### What's the use case?

When we launch a backfill for a selection of assets using the GraphQL API `launchPartitionBackfill`, it creates a backfill and dagster that will launch runs for the targeted assets and their targeted partitions consequentially. 

However, if something went wrong or we just want to cancel the entire backfill operation right after we launch it, there is no way to cancel the entire backfill all at once.

It seems like the only option to terminate computation for the backfill is to use the "Terminate unfinished runs"  button on the backfill overview page. However, it seems like it is only terminating whatever runs in progress and all queued/requested runs that had not been launched are not canceled. 

![image](https://github.com/dagster-io/dagster/assets/109117963/176a60f3-e0d1-403c-b354-3e9326068a9e)

Basically, we need to keep doing the "terminate unfinished runs" to terminate everything for a backfill, this can be error-prone and a lot of unnecessary waste of resources can occur. 

### Ideas of implementation

Along with the "Terminate unfinished runs", also provide a button to "Cancel backfill" which internally will terminate all the in-progress runs and "queued" runs, which effectively cancel out all the computation and queuing created for the backfill. 

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@clairelin135 - mind taking a look at this one? It might be related or duplicative with what you're currently working on?

**Comment by clairelin135** [comment]
Hi @ruizh22. Not sure if I'm missing something here--to my knowledge we have:
- a "cancel backfill submission" button, which prevents more runs from being created
- a "terminate unfinished runs" button, which cancels all of the runs that are queued/started

These two buttons together should give you what you're looking for? Wondering if I'm missing something or that "cancel backfill" button isn't displaying for you.

<img width="1261" alt="image" src="https://github.com/dagster-io/dagster/assets/29110579/6892010a-b23d-4eef-8b5a-361955dc11d2">

**Comment by ruizh22** [comment]
hi @clairelin135 , sorry for the delay.

Yes, the "cancel backfill submission" button isn't displaying for me right now. We are on 1.3.2 atm. 
![image](https://github.com/dagster-io/dagster/assets/109117963/347152df-857c-4dbd-8eb2-23c197dae3f0)

**Comment by clairelin135** [comment]
@ruizh22 I think this issue only happens for backfills that target an asset selection, correct? In this week's release (1.3.7) we will support canceling these asset backfills.

These backfills will now display a "cancel backfill submission" button, which when clicked will prevent future runs from being queued and mark unfinished runs as canceling.

**Comment by ruizh22** [comment]
hi @clairelin135, that's awesome! Will keep an eye out for the 1.3.7 👍
